packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js

```diff
@@ -2109,22 +2099,164 @@ describe('Checkout', () => {
                     }
                 });
 
-                const payload = {
-                    action: 'submit'
-                };
+                // Act & Assert
+                expect(wrapper.vm.onFormSubmit).toBeDefined();
+            });
 
-                // Act
-                await wrapper.vm.onFormSubmit();
+            describe('when invoked', () => {
+                it('should make a call to `trackFormInteraction` so we can track the action type `submit`', async () => {
+                    // Arrange
+                    isFormValidSpy.mockReturnValue(true);
+                    const trackFormInteractionSpy = jest.spyOn(VueCheckout.methods, 'trackFormInteraction');
+                    const wrapper = mount(VueCheckout, {
+                        store: createStore(),
+                        i18n,
+                        localVue,
+                        propsData,
+                        mocks: {
+                            $v,
+                            $logger
+                        }
+                    });
 
-                // Assert
-                expect(trackFormInteractionSpy).toHaveBeenCalledWith(payload);
+                    // Act
+                    await wrapper.vm.onFormSubmit();
+
+                    // Assert
+                    expect(trackFormInteractionSpy).toHaveBeenCalledWith({
+                        action: 'submit'
+                    });
+                });
+
+                describe('AND the call to `isFormValid` is falsey', () => {
+                    it('should call `onInvalidCheckoutForm` so we can emit, track and log error information', async () => {
+                        // Arrange
+                        isFormValidSpy.mockReturnValue(false);
+                        const mockValidationState = {
+                            validFields: [
+                                'customer.mobileNumber',
+                                'address.line1',
+                                'address.city',
+                                'address.postcode'
+                            ],
+                            invalidFields: []
+                        };
+
+                        const getFormValidationStateSpy = jest.spyOn(validations, 'getFormValidationState');
+                        getFormValidationStateSpy.mockReturnValue(mockValidationState);
```

Comment:
We could replace this with:
`jest.spyOn(validations, 'getFormValidationState').mockReturnValue(mockValidationState);`, given we're not using this variable anywhere, unless I'm mistaken.

Reply:
Fixed.